From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sun, 30 Jul 2023 16:38:30 +0200
Subject: [PATCH] Client perspective - Item replacements - Send replaceable
 updates after game mode change

License: Fiddle Public License Agreement 1.0 (included in license/FPL-1.0.txt)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
index 34ecfb89372f459117db99d57a7edd6f681bbe8a..dacaf46e947a18dcae4427d57a3d905bc3434f8b 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -103,6 +103,15 @@ public class ServerPlayerGameMode {
     protected void setGameModeForPlayer(GameType gameMode, @Nullable GameType previousGameMode) {
         this.previousGameModeForPlayer = previousGameMode;
         this.gameModeForPlayer = gameMode;
+        // Fiddle start - client perspective - item replacements - send update packets for replaceables after game mode change
+        if (this.player.hasFinishedBeingPlacedIntoPlayerList) {
+            if (previousGameMode != gameMode) {
+                if (previousGameMode == GameType.CREATIVE || gameMode == GameType.CREATIVE) {
+                    this.player.sendUpdatesForDeepReplaceableContent(true, false);
+                }
+            }
+        }
+        // Fiddle end - client perspective - item replacements - send update packets for replaceables after game mode change
         gameMode.updatePlayerAbilities(this.player.getAbilities());
     }
 
